### ISSUE TITLE
Another SQL False positive

### ISSUE THREAD

**Issue by phene** [presidentbeef/brakeman#655]
# String literals with no interpolation should not be treated as SQL injection.
## Error

```
High Confidence
Possible SQL injection near line 48: joins(:things).joins("LEFT JOIN thing_memberships ON items.thing_id = thing_memberships.thing_id").where(:tenant_id => user.tenant_id).where(Thing.member_or_visibility(user, [Thing::VISIBILITY_PUBLIC, Thing::VISIBILITY_COMPANY]))
```
## Code

``` ruby
class Thing
  scope :viewable_by, ->(user) {
    member_join = 'LEFT JOIN thing_memberships ON items.thing_id = thing_memberships.thing_id'
    viewable_thing = Thing.member_or_visibility(user, [Thing::VISIBILITY_PUBLIC, Thing::VISIBILITY_COMPANY])
    joins(:things).joins(member_join).where(:tenant_id => user.tenant_id).where(viewable_thing).distinct
  }
end
```

**Comment by presidentbeef** [comment]
The warning is on `Thing.member_or_visibility`, not the string literal.

**Comment by phene** [comment]
Why does it assume `member_or_visiblity` returns something unsafe?

**Comment by presidentbeef** [comment]
Because anything potentially coming from the database is considered unsafe. In this case `Thing` must be an ActiveRecord model. It doesn't know anything about `member_or_visiblity`.

However, I do agree this should probably be bumped down to a weak confidence warning.

**Comment by presidentbeef** [comment]
This is fixed by #985
